TheraLang.BLL/Services/MemberFeeService.cs

```diff
@@ -41,24 +41,15 @@ public async Task DeleteAsync(int id)
 
                 _unitOfWork.Repository<MemberFee>().Remove(memberFee);
                 await _unitOfWork.SaveChangesAsync();
-            }
-            catch (Exception e)
-            {
-                e.Data["Id"] = id;
-                throw;
-            }
         }
 
         public async Task AddAsync(MemberFeeDto memberFeeDto)
         {
-            try
-            {
-                if (memberFeeDto == null)
+                var minDate = getMinDate();
+                if (memberFeeDto.FeeDate <= minDate)
                 {
-                    throw new NullReferenceException(
-                        $"{nameof(MemberFeeDto)} cannot be null");
+                    throw new InvalidArgumentException(nameof(MemberFeeDto.FeeDate), "Invalid FeeDate");
```

Comment:
I think this message can be more informative, and you probably should see on InvalidArgumentException class and make your message appropriate to it.
public InvalidArgumentException(string nameOfArg, string reason) : base("Invalid parameter: {0}, reason: {1}",
            nameOfArg, reason)

Reply:
Done